pymatgen/core/trajectory.py

```diff
@@ -542,15 +544,30 @@ def from_file(cls, filename: str | Path, constant_lattice: bool = True, **kwargs
             structures = Xdatcar(filename).structures
         elif fnmatch(fname, "vasprun*.xml*"):
             structures = Vasprun(filename).structures
+        elif fnmatch(fname, "*.traj"):
+            ase_traj = ASE_Trajectory(fname)
+            pbc = ase_traj[0].pbc
+            if any(pbc):
+                structures = [AseAtomsAdaptor.get_structure(atoms) for atoms in ase_traj]
+            else:
+                molecules = [AseAtomsAdaptor.get_molecule(atoms) for atoms in ase_traj]
+
         else:
             supported = ("XDATCAR", "vasprun.xml")
             raise ValueError(f"Expect file to be one of {supported}; got {filename}.")
 
-        return cls.from_structures(
-            structures,
-            constant_lattice=constant_lattice,
-            **kwargs,
-        )
+        if "structures" in locals():
```

Comment:
Using `locals()` here is unnecessary. Cheaper to define `is_struct_traj = any(ase_traj[0].pbc)` and return based on that.

Reply:
Here I think the reason I use `locals()` is because `from_file()` can also read other file formats (`XDATCAR` and `vasprun.xml`), and `any(ase_traj[0].pbc)` does not apply to those cases. Is there a better way to handle this?

Reply:
Using `locals` seems too brittle. How about you call it `structures` either way and define  `is_mol = False` at the top of the method and then modify it later as needed?

Reply:
Thanks! Just commited a new update accordingly. (The name should be "Add flag `is_mol`", somehow git doesn't recognize the quotation marks)